src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -256,7 +270,12 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
                     
                     if (cachedData != null)
                     {
-                        data = JsonSerializer.Deserialize<IDictionary<string, ConfigurationSetting>>(cachedData);
+                        IDictionary<string, ConfigurationSetting> offlineCacheData = JsonSerializer.Deserialize<IDictionary<string, ConfigurationSetting>>(cachedData);
+
+                        foreach (KeyValuePair<string, ConfigurationSetting> setting in offlineCacheData)
+                        {
+                            data[setting.Key] = new List<ConfigurationSetting>() { setting.Value };

```

Comment:
>data [](start = 28, length = 4)

Have we validated this scenario? Where is the dictionary being initialized?

Reply:
Initialized and added test for validating this scenario.